src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -181,31 +206,81 @@ private static void verifyLabel(Object label) {
      *     ...
      * }
      */
-    private static MethodHandle createRepeatIndexSwitch(MethodHandles.Lookup lookup, Object[] labels) {
-        MethodHandle def = MethodHandles.dropArguments(MethodHandles.constant(int.class, labels.length), 0, Object.class);
+    private static MethodHandle createRepeatIndexSwitch(MethodHandles.Lookup lookup, Object[] labels, Class<?> selectorType) {
+        MethodHandle def = MethodHandles.dropArguments(MethodHandles.constant(int.class, labels.length), 0, selectorType);
         MethodHandle[] testChains = new MethodHandle[labels.length];
         List<Object> labelsList = List.of(labels).reversed();
-
+        // unconditionally exact patterns always match
+        MethodHandle trueDef = MethodHandles.dropArguments(MethodHandles.constant(boolean.class, true), 0, selectorType, Object.class);
         for (int i = 0; i < labels.length; i++) {
             MethodHandle test = def;
             int idx = labels.length - 1;
             List<Object> currentLabels = labelsList.subList(0, labels.length - i);
-
             for (int j = 0; j < currentLabels.size(); j++, idx--) {
                 Object currentLabel = currentLabels.get(j);
+                Object testLabel = currentLabel;
                 if (j + 1 < currentLabels.size() && currentLabels.get(j + 1) == currentLabel) continue;
                 MethodHandle currentTest;
-                if (currentLabel instanceof Class<?>) {
-                    currentTest = INSTANCEOF_CHECK;
-                } else if (currentLabel instanceof Integer) {
-                    currentTest = INTEGER_EQ_CHECK;
-                } else if (currentLabel instanceof EnumDesc) {
+                if (currentLabel instanceof Class<?> currentLabelClass) {
+                    if (unconditionalExactnessCheck(selectorType, currentLabelClass)) {
+                        currentTest = trueDef;
+                    } else if (currentLabelClass.isPrimitive()) {
+                         if (selectorType.equals(Object.class)) {
+                            currentTest = INSTANCEOF_CHECK;
+                            if (currentLabelClass.isAssignableFrom(byte.class)) { testLabel = Byte.class; }
```

Comment:
I assume here you want to box the primitive, right? If that's the case, consider using the JDK class `sun.invoke.util.Wrapper`, which has a factory `forBasicType` that takes a class. From there you can get the boxed type easily, in a single line.

Reply:
Perfect! Exactly what I needed.